services/bettersocial/fcmToken/sendChatNotification.js

```diff
@@ -4,14 +4,25 @@ const {FcmToken, User} = require('../../../databases/models');
 const UsersFunction = require('../../../databases/functions/users');
 
 const sendChatNotification = async (userTargetId, payload) => {
-  const signedUser = await UsersFunction.findSignedUserId(User, userTargetId);
-  const userTargetToken = await FcmTokenFunction.findAllTokenByUserId(FcmToken, signedUser);
+  const signedUser = await UsersFunction.findSignedUserId(User, userTargetId, {
+    shouldReturnObject: true
+  });
+
+  const userTargetToken = await FcmTokenFunction.findAllTokenByUserId(
+    FcmToken,
+    signedUser?.user_id
+  );
+
   if (userTargetToken) {
     try {
       userTargetToken.forEach((user) => {
         const newPayload = {
           ...payload,
-          token: user?.token
+          token: user?.token,
+          data: {
+            ...payload.data,
+            is_annoymous: signedUser?.is_user_anonymous?.toString()
+          }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The payload for the messaging service has been extended to include `is_anonymous` from the `signedUser` object. However, there's a typo in the property name (`is_annoymous` instead of `is_anonymous`). Also, consider what happens if `signedUser?.is_user_anonymous` is undefined - you're converting it to a string, which would result in the string "undefined". You may want to provide a default value in this case.

```diff
- is_annoymous: signedUser?.is_user_anonymous?.toString()
+ is_anonymous: (signedUser?.is_user_anonymous ?? false).toString()
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->